frontend/src/components/NodePathHistory/NodePathHistory.jsx

```diff
@@ -0,0 +1,167 @@
+
+/**
+ * Join the list of node names into one string
+ */
+function convertPathToString(path) {
+  return path.join("/");
+}
+
+export const NodePathHistory = ({
+  path,
+  onNodeNameClick,
+  onPathChange,
+  isValidPath,
+}) => {
+  const [mode, setMode] = useState(MODES.VIEW);
+  const [currentText, setCurrentText] = useState("");
+  const [errorOpen, setErrorOpen] = React.useState(false);
+
+  useEffect(() => {
+    setCurrentText(convertPathToString(path));
+  }, [path]);
+
+  /**
+   * Store the current text being typed
+   */
+  const onTextChangeHandler = (event) => {
+    setCurrentText(event.target.value);
+  };
+
+  /**
+   * Triggered when the user is finished editing the path
+   * and clicks done. Notifies the parent of the change
+   * through a callback, passing in the new path as a parameter.
+   */
+  const onDoneClickHandler = () => {
+    // Remove preceding or trailing slashes and spaces
+    const newPath = currentText
+      .trim()
+      .replace(/^\/+/, "")
+      .replace(/\/+$/, "")
+      .split("/");
+
+    // Inform the parent container of the change, if there was one and
+    // given that the path is valid
+    if (currentText !== convertPathToString(path)) {
+      if (isValidPath(newPath)) {
+        onPathChange(newPath);
+      } else {
+        setErrorOpen(true);
+        clearChanges();
+      }
+    }
+  };
+
+  /**
+   * If the user clicks outside of the text field, reset
+   * the text inside of it and go back to view mode.
+   */
+  const clearChanges = () => {
+    setCurrentText(convertPathToString(path));
+    setMode(MODES.VIEW);
+  };
+
+  /**
+   * This function is called when the alert close button
+   * is clicked or there is a timeout.
+   */
+  const handleClose = (event, reason) => {
+    // Should not close if the user just clicks anywhere
+    if (reason === "clickaway") {
+      return;
+    }
+    setErrorOpen(false);
+  };
+
+  const Links = path.slice(0, -1).map((name, index) => (
+    <Button key={index} onClick={() => onNodeNameClick(name)}>
+      <Box>{name}</Box>
+    </Button>
+  ));
+
+  const BreadcrumbsView = (
+    <Breadcrumbs
+      maxItems={4}
+      itemsAfterCollapse={3}
+      itemsBeforeCollapse={0}
+      color="primary"
+      aria-label="breadcrumb"
+      separator={<NavigateNext fontSize="small" />}
+    >
+      {Links}
+
+      {/* The last element is styled differently and not rendered as a link */}
+      <Button disableRipple>
+        <Box>{path[path.length - 1]}</Box>
+      </Button>
+    </Breadcrumbs>
+  );
+
+  const EditPathView = (
+    <ClickAwayListener onClickAway={clearChanges}>
+      <TextField
+        label="Path"
+        value={currentText}
+        variant="outlined"
+        size="small"
+        onChange={onTextChangeHandler}
+        onKeyDown={(e) => {
+          // If the enter key is pressed, it is equivalent
+          // to clicking on Done
+          if (e.keyCode === 13) onDoneClickHandler();
+        }}
+        fullWidth
```

Comment:
Just to confirm, this full width means the width of the container right? Just want to make sure that it is compatible with mine and Sultan's MR, which opens the textbox on the right side

Reply:
Ya, it's just the container, so it shouldn't interfere with anything else.